### ISSUE TITLE
Reduce non-core deps slightly

### ISSUE THREAD

**Issue by preaction** [preaction/Statocles#255]
Reducing our non-core deps will improve our install time and the success rate (however slightly). This makes it easier to bundle the app (#2, #6)

We don't want to completely remove non-core deps, there's no good reason not to use:

* Moo
* Type::Tiny
* Path::Tiny
* Import::Base

And some modules are core to our functionality like:

* Beam::Wire
* YAML
* Git::Repository

But some modules are trivial to implement in Perl, and others are already basically provided by Mojolicious:

* File::Copy::Recursive is a few lines, given we already rely on Path::Tiny
* The things in List::MoreUtils are probably already in Mojo::Collection
* What we use of Types::Path::Tiny could be simply added to Statocles::Types

See the commit message for b94550ab7a7ebe8204e7ddc4dd64fa5c42753df6

In other words, we should add new dependencies liberally, but also remove them liberally.

**Comment by preaction** [comment]
Look at Mojo::Util, there are things in there we could use to replace some bits.

Also, YAML::Tiny might do everything we need it to, and more consistently than YAML.pm...
